applications/FluidDynamicsApplication/python_scripts/apply_outlet_process.py

```diff
@@ -68,26 +80,63 @@ def ExecuteInitializeSolutionStep(self):
+        # Compute the outlet average velocity
+        self._ComputeOutletCharacteristicVelocity()
 
 
     def ExecuteFinalizeSolutionStep(self):
         # Call the base process ExecuteFinalizeSolutionStep()
         self.aux_pressure_process.ExecuteFinalizeSolutionStep()
         self.aux_external_pressure_process.ExecuteFinalizeSolutionStep()
+
+
+    # Private methods section
+    def _AddOutletHydrostaticComponent(self):
+        # Get the body force value
+        for node in self.outlet_model_part.Nodes:
+            body_force = node.GetSolutionStepValue(KratosMultiphysics.BODY_FORCE, 0)
+            break
+
+        # Compute the body force unit normal vector
+        body_force_norm = math.sqrt(math.pow(body_force[0],2)+math.pow(body_force[1],2)+math.pow(body_force[2],2))  # Body force norm
+        body_force_dir = (1/(body_force_norm+1e-10))*body_force                                                     # Body force unit director vector
+
+        # Compute the minimum body force projection value (reference value)
+        min_proj = 0.0
+        for node in self.outlet_model_part.Nodes:
+            body_force_proj = body_force_dir[0]*node.X + body_force_dir[1]*node.Y + body_force_dir[2]*node.Z    # Iteration node body force projection
+            min_proj = min(min_proj, body_force_proj)
+
+        # Add the hydrostatic component to the current PRESSURE and EXTERNAL_PRESSURE values
+        for node in self.outlet_model_part.Nodes:
+            body_force_proj = body_force_dir[0]*node.X + body_force_dir[1]*node.Y + body_force_dir[2]*node.Z    # Iteration node body force projection
+            rho = node.GetSolutionStepValue(KratosMultiphysics.DENSITY, 0)                                      # Nodal density value
+            hyd_pres = rho*body_force_norm*(self.h_top + (body_force_proj-min_proj))                            # Iteration node hydrostatic pressure
+            cur_pres = node.GetSolutionStepValue(KratosMultiphysics.EXTERNAL_PRESSURE, 0)                       # Iteration node imposed external pressure
+
+            node.SetSolutionStepValue(KratosMultiphysics.PRESSURE, 0, hyd_pres+cur_pres)                        # Add the hydrostatic component to the PRESSURE value
+            node.SetSolutionStepValue(KratosMultiphysics.EXTERNAL_PRESSURE, 0, hyd_pres+cur_pres)               # Add the hydrostatic component to the EXTERNAL_PRESSURE value
+
+
+    def _ComputeOutletCharacteristicVelocity(self):
+        # Compute the current process average velocity
+        outlet_avg_vel_norm = 0.0
+        for node in self.outlet_model_part.GetCommunicator().LocalMesh().Nodes:
+            vnode = node.GetSolutionStepValue(KratosMultiphysics.VELOCITY)
+            outlet_avg_vel_norm += math.sqrt(vnode[0]**2+vnode[1]**2+vnode[2]**2)
+            outlet_avg_vel_norm /= len(self.outlet_model_part.GetCommunicator().LocalMesh().Nodes)
```

Comment:
After a short research, I found out that for low values of "n" naive product x*x is cheaper than x**n or pow(x,n). Let me change it before merging.